nailgun-examples/src/main/java/com/martiansoftware/nailgun/examples/Hash.java

```diff
@@ -18,93 +18,93 @@
 
 package com.martiansoftware.nailgun.examples;
 
+import com.martiansoftware.nailgun.NGContext;
 import java.security.MessageDigest;
 import java.security.Provider;
 import java.security.Security;
 import java.util.Iterator;
 import java.util.Set;
 
-import com.martiansoftware.nailgun.NGContext;
-
 /**
- * Hashes the client's stdin to the client's stdout in the form of
- * a hexadecimal string.  Command line requires one parameter: either the name
- * of the algorithm to use (e.g., "MD5"), or "?" to request a list of
- * available algorithms.
- * 
+ * Hashes the client's stdin to the client's stdout in the form of a hexadecimal string. Command
+ * line requires one parameter: either the name of the algorithm to use (e.g., "MD5"), or "?" to
+ * request a list of available algorithms.
+ *
  * @author <a href="http://www.martiansoftware.com/contact.html">Marty Lamb</a>
  */
 public class Hash {
-	
-	// used to turn byte[] to string
-	private static final char[] HEXCHARS = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'}; 
-
-	/**
-	 * Provides a list of algorithms for the specified service (which, for
-	 * our purposes, is "MessageDigest".
-	 * 
-	 * This method was only very slightly adapted (to use a TreeSet) from
-	 * the Java Almanac at http://javaalmanac.com/egs/java.security/ListServices.html 
-	 * @param serviceType The name of the service we're looking for.  It's "MessageDigest"
-	 */
-    private static Set getCryptoImpls(String serviceType) {
-        Set result = new java.util.TreeSet();
-    
-        // All all providers
-        Provider[] providers = Security.getProviders();
-        for (int i=0; i<providers.length; i++) {
-            // Get services provided by each provider
-            Set keys = providers[i].keySet();
-            for (Iterator it = keys.iterator(); it.hasNext(); ) {
-                String key = (String)it.next();
-                key = key.split(" ")[0];
-    
-                if (key.startsWith(serviceType+".")) {
-                    result.add(key.substring(serviceType.length()+1));
-                } else if (key.startsWith("Alg.Alias."+serviceType+".")) {
-                    // This is an alias
-                    result.add(key.substring(serviceType.length()+11));
-                }
-            }
+
+  // used to turn byte[] to string
+  private static final char[] HEXCHARS = {
+    '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'
+  };
+
+  /**
+   * Provides a list of algorithms for the specified service (which, for our purposes, is
+   * "MessageDigest".
+   *
+   * <p>This method was only very slightly adapted (to use a TreeSet) from the Java Almanac at
+   * http://javaalmanac.com/egs/java.security/ListServices.html
+   *
+   * @param serviceType The name of the service we're looking for. It's "MessageDigest"
+   */
+  private static Set getCryptoImpls(String serviceType) {
+    Set result = new java.util.TreeSet();
```

Comment:
nit: why fully qualified import? Does it conflict with something?

Reply:
also, why is it not generic? :)

Reply:
Only code formatting changes were in 'examples' folder